mr/joiner_executor.cc

```diff
@@ -173,7 +192,8 @@ void JoinerExecutor::ProcessInputQ(detail::TableBase* tb) {
 
       SetFileName(is_binary, ii.fspec->url_glob(), raw_context);
       SetMetaData(*ii.fspec, raw_context);
-      per_io_->items_cnt += runner_->ProcessInputFile(ii.fspec->url_glob(), ii.wf->type(), emit_cb);
+      cnt += runner_->ProcessInputFile(ii.fspec->url_glob(), ii.wf->type(), emit_cb);
+      raw_context->IncBy("fn-calls", cnt);
```

Comment:
you have a bug here i think.
you adding a quadratic expression.
Please add a check in one of the tests in mr_test in case I am right.